docs(tutorials): focalboard (#1339)
scaleway/docs-content: tutorials/focalboard-project-management/index.mdx

---
meta:
  title: Project management for technical teams with Focalboard on Ubuntu Instances
  description: This page shows how to setup a Focalboard project-management tool on Cost-Optimzed Instances running Ubuntu Linux
content:
  h1: Project management for technical teams with Focalboard on Ubuntu Instances
  paragraph: This page shows how to setup a Focalboard project-management tool on Cost-Optimzed Instances running Ubuntu Linux
tags: compute instance focalboard ubuntu project-management open-source
hero:
categories:

>   - compute

Reviewer: ici on a compute en catégorie mais est-ce qu'on rajoute pas use-cases ?

Author: il y a build and maintain

  - instance
  - build-and-maintain
dates:
  validation: 2023-02-22
  posted: 2023-02-22
---

> Focalboard is an open-source, multilingual, self-hosted project management tool developed by the team around Mattermost. 

Reviewer: ```suggestion
Focalboard is an open-source, multilingual, self-hosted project management tool developed by Mattermost. 
```

> It helps organizations to maintain a single source of truth, so teams stay aligned to complete tasks, reach milestones, and achieve their goals. 

Author: ```suggestion
With Focalboard, you'll never lose sight of the big picture. This incredible tool helps you maintain a single source of truth, so your team can stay aligned and focused on completing tasks, reaching milestones, and achieving their goals. And with a full suite of features at your fingertips, including Kanban, table, gallery, and calendar views, you can easily filter and prioritize tasks to stay focused on what matters most.
```

> It helps your organization maintain a single source of truth so teams stay aligned to complete tasks, reach milestones, and achieve their goals. 

Author: ```suggestion
It helps organizations to maintain a single source of truth, so teams stay aligned to complete tasks, reach milestones, and achieve their goals. 
```

> Focalboard provides a full suite of features such as Kanban, table, gallery, and calendar view. Filters help you to stay focussed on the highest priority items. 

Author: ```suggestion
Focalboard provides a full suite of features such as Kanban, table, gallery, and calendar view. Filters allow staying focussed on the highest priority items. 
```

The application is a privacy-caring alternative to Trello, Notion, and Asana and can be installed easily on Scaleway Instances.

<Message type="tip">
  We recommend you follow this tutorial using a [Cost-Optimized Instance](/compute/instances/reference-content/choosing-instance-type/).
</Message>

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [configured your SSH Key](/console/my-project/how-to/create-ssh-key/)
  - You have [created an Instance](/compute/instances/how-to/create-an-instance/) that runs Ubuntu Jammy Jellyfish (22.04 LTS)
  - You have a (sub)domain pointed to the IP address of your Instance (A-record)
</Message>

## Installation of the required software

> In the following steps you make sure that the software installed on your Instance is up-to-date, you install the required software packages and download and unpack the Focalboard application.

Reviewer: ```suggestion
In the following steps, we will make sure that the software and the required packages installed on your Instance are up-to-date, and that you have downloaded and unpacked the Focalboard application.
```

1. Log into your Instance using SSH:
    ```
    ssh root@<your_scaleway_instance_ip>
    ```
2. Update the `apt` package manager and upgrade the installed software on your Instance to the latest version.
    ```
    apt update && apt upgrade -y
    ```
3. Install the required software packages using the `apt` package manager.
    ```
    apt install nginx python3-certbot-nginx mariadb-server ufw
    ```
4. Download and unpack Focalboard. In this tutorial, we use v7.8.2 of Focalboard, but you can find the **lates**t release in the project's [GitHub repository](https://github.com/mattermost/focalboard/releases).
    ```
    wget https://github.com/mattermost/focalboard/releases/download/v7.8.2/focalboard-server-linux-amd64.tar.gz
    tar -xvzf focalboard-server-linux-amd64.tar.gz
    mv focalboard /opt
    ```

## Nginx configuration

> We use Nginx aas a proxy in front of the application to prevent it from direct access and by providing an additional layer of security through TLS encryption.

Reviewer 2: ```suggestion
We use Nginx as a proxy in front of the application to prevent it from direct access and by providing an additional layer of security through TLS encryption.
```

        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Frame-Options SAMEORIGIN;
        proxy_buffers 256 16k;
        proxy_buffer_size 16k;
        proxy_read_timeout 600s;
        proxy_cache_revalidate on;
        proxy_cache_min_uses 2;
        proxy_cache_use_stale timeout;
        proxy_cache_lock on;
        proxy_http_version 1.1;
        proxy_pass http://focalboard;
    }
    }
    ```
2. Test the new configuration, enable the site, and reload Nginx to activate the Focalboard site.
    ```
    nginx -t
    ln -s /etc/nginx/sites-available/focalboard.example.com.conf /etc/nginx/sites-enabled/focalboard.example.com.conf
    service nginx reload
    ```
3. Run `certbot` to obtain a TLS certificate for your domain name and let the application configure Nginx automatically.
    ```
    certbot
    ```
    <Message type="tip">
      For more details on using [Certbot](https://certbot.eff.org/), refer to our [dedicated documentation](/tutorials/nginx-reverse-proxy/#adding-tls-to-your-nginx-reverse-proxy-using-let's-encrypt).
    </Message>

## MariaDB configuration

> Focalboard uses a MariaDB database to store it's data. Initialize the database engine, create a table and user for Focalboard.

Reviewer: ```suggestion
Focalboard uses a MariaDB database to store its data. Initialize the database engine, and create a table and user for Focalboard.
```

    ```
    Follow the steps in the wizard:
    * Enter current password for root: Press **Enter** to continue.
    * Set root password? **Y**
    * Create and confirm a new password
    * Remove anonymous users? **Y**
    * Disallow root login remotely? **Y**
    * Remove test database and access to it? **Y**
    * Reload privilege tables now? **Y**
2. Log into the MariaDB shell using the MySQL user `root` and the password created in the previous step.
    ```
    mysql -u root -p
    ```
3. Create a table for Focalboard and grant permission to a MariaDB user. Change the values for `focalboarduser` and `focalboardpassword` to your own user/password combination.
    ```
    CREATE DATABASE focalboard;
    GRANT ALL on focalboard.* to 'focalboarduser'@'localhost' identified by 'focalboardpassword';
    ```
    Then leave the MariaDB shell by typing `exit;`.
4. Open the Focalboard configuration file in a text editor.
    ```
    nano /opt/focalboard/config.json
    ```
    Edit the database configuration as shown below:
    ```
    "dbtype": "mysql",
    "dbconfig": "focalboarduser:focalboardpassword@tcp(127.0.0.1:3306)/focalboard",
    ```

## Configure a Focalboard service

> We configure a service that will will keep the Focalboard server running across reboots.

Reviewer 2: ```suggestion
We are now configuring a service that will keep the Focalboard server running across reboots.
```

1. Create a service description. 
    ```
    nano /lib/systemd/system/focalboard.service
    ```
    Copy the following content into the file, save it and exit the text editor.
2. Reload the systemd configuration, enable and start the Focalboard service.
    ```
    systemctl daemon-reload
    systemctl start focalboard.service
    systemctl enable focalboard.service
    ```

## Firewall configuration

> To avoid the Focalboard application being directly reachable via plain HTTP on port `8000`, we configure [a firewall](/tutorials/installation-uncomplicated-firewall/) to block all incoming ports, except HTTP, HTTPS, and SSH.

Reviewer 2: ```suggestion
To avoid the Focalboard application being directly reachable via plain HTTP on port `8000`, we will configure [a firewall](/tutorials/installation-uncomplicated-firewall/) to block all incoming ports, except HTTP, HTTPS, and SSH.
```

> To avoid the Focalboard application being directly reachable via plain HTTP on Port 8000, we configure [a firewall](/tutorials/installation-uncomplicated-firewall/) to block all incoming ports, except HTTP, HTTPS, and SSH.

Reviewer: ```suggestion
To avoid the Focalboard application being directly reachable via plain HTTP on port `8000`, we configure [a firewall](/tutorials/installation-uncomplicated-firewall/) to block all incoming ports, except HTTP, HTTPS, and SSH.
```
